nominal/core/connection.py

```diff
@@ -71,6 +75,15 @@ def get_write_stream(
             max_wait: How long a batch can exist before being flushed to Nominal.
             data_format: Serialized data format to use during upload.
                 NOTE: selecting 'protobuf' requires that `nominal` was installed with `protos` extras.
+            file_fallback: Filepath to write failed batches to during streaming
+                NOTE: expects a .avro filename
+                NOTE: only works with `data_format='rust_experimental'`
+            log_level: Log level to use in underlying rust streaming code.
+                NOTE: Should be a rust log level e.g. 'debug', 'trace', 'info', etc.
+                NOTE: only works with `data_format='rust_experimental'`
+            num_workers: Number of worker threads to use in underlying rust streaming code.
+                NOTE: use with care-- this may have large impacts on streaming performance.
+                NOTE: only works with `data_format='rust_experimental'`
```

Comment:
typing.overload will allow you to type this so that the log level, file fallback, and num_workers will only appear on the signature if using "rust_experimental"